pkg/tests/tasks/security/authorization/http_test.go

```diff
@@ -16,105 +16,230 @@ package authorization
-			log.Log.Errorf("GET policy failed. Got response: %s", string(body))
-		} else {
-			log.Log.Infof("Got expected page.")
-		}
-		util.CloseResponseBody(resp)
+func TestAuthoriztionHTTPTraffic(t *testing.T) {
+	test.NewTest(t).Id("T20").Groups(test.Full, test.ARM, test.InterOp).Run(func(t test.TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+
+		ns := "bookinfo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+			shell.Execute(t,
+				fmt.Sprintf(`kubectl patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"dataPlane":{"mtls":false},"controlPlane":{"mtls":false}}}}'`,
+					meshNamespace, smcpName))
+			shell.Execute(t,
+				fmt.Sprintf(`kubectl -n %s wait --for condition=Ready smcp/%s --timeout 180s`,
+					meshNamespace, smcpName))
+		})
+
+		t.Log("This test validates authorization policies for HTTP traffic.")
+		t.Log("Doc reference: https://istio.io/v1.14/docs/tasks/security/authorization/authz-http/")
+
+		t.LogStep("Enable Service Mesh Control Plane mTLS")
+		shell.Execute(t,
+			fmt.Sprintf(`kubectl patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"dataPlane":{"mtls":true},"controlPlane":{"mtls":true}}}}'`,
+				meshNamespace, smcpName))
+		shell.Execute(t,
+			fmt.Sprintf(`kubectl -n %s wait --for condition=Ready smcp/%s --timeout 180s`,
+				meshNamespace, smcpName))
+
+		t.LogStep("Install bookinfo with mTLS")
+		app.InstallAndWaitReady(t, app.BookinfoWithMTLS(ns))
+		GATEWAY_URL := shell.Execute(t,
+			fmt.Sprintf(`kubectl get routes -n %s istio-ingressgateway -o jsonpath='{.spec.host}'`, meshNamespace))
+
+		t.NewSubTest("deny all http traffic to bookinfo").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, DenyAllPolicy)
+			})
+			t.LogStep("Apply policy that denies all HTTP requests to bookinfo workloads")
+			oc.ApplyString(t, ns, DenyAllPolicy)
+
+			t.LogStep("Verify that GET request is denied")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				shell.Execute(t,
+					fmt.Sprintf(`curl http://%s/productpage`, GATEWAY_URL),
+					assert.OutputContains(
+						"RBAC: access denied",
+						"ProductPage returns expected message: 'RBAC: access denied'",
+						"ProductPage didn't return 'RBAC: access denied'"),
+				)
+			})
+		})
+
+		t.NewSubTest("only allow HTTP GET request to the productpage workload").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.DeleteFromString(t, ns, ProductpageGETPolicy)
+				oc.DeleteFromString(t, ns, DenyAllPolicy)
+			})
+			t.LogStep("Apply policies that allow access with GET method to the productpage workload and deny requests to other workloads")
+			oc.ApplyString(t, ns, DenyAllPolicy)
+			oc.ApplyString(t, ns, ProductpageGETPolicy)
+
+			t.LogStep("Verify that GET request to the productpage is allowed and fetching other services is denied")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				shell.Execute(t,
+					fmt.Sprintf(`curl http://%s/productpage`, GATEWAY_URL),
```

Comment:
You can replace this with `app.BookinfoProductPageURL()` (best to store it in a variable, since it runs a shell command)